scripts/master.sh

```diff
@@ -1,13 +1,22 @@
 #!/usr/bin/bash
 set -eu
 
+PRIVATE_IP=""
+while [ -z $PRIVATE_IP ]
+do
+  sleep 1
+  PRIVATE_IP=$(ifconfig | grep -A 1 'ens10' | tail -1 | awk '{print $2}'| cut -d ':' -f 2)
+done
+PUBLIC_IP=$(ifconfig | grep -A 1 'eth0' | tail -1 | awk '{print $2}'| cut -d ':' -f 2)
 # Initialize Cluster
+
 if [[ -n "$FEATURE_GATES" ]]
 then
-  kubeadm init --pod-network-cidr=192.168.0.0/16 --feature-gates "$FEATURE_GATES"
+  kubeadm init --apiserver-advertise-address $PRIVATE_IP --pod-network-cidr=192.168.0.0/16 --apiserver-cert-extra-sans $PUBLIC_IP --feature-gates $FEATURE_GATES
 else
-  kubeadm init  --pod-network-cidr=192.168.0.0/16
+  kubeadm init --apiserver-advertise-address $PRIVATE_IP --pod-network-cidr=192.168.0.0/16 --apiserver-cert-extra-sans $PUBLIC_IP
```

Comment:
hey @solidnerd 
I currently struggle on this part. I found this option `--apiserver-advertise-address` and this made it work that the nodes speak over the private network. 
But now also the downloaded admin.conf has the private IP in there, so it does not work to connect from the local machine.  
My naive approach would be a `sed` to replace the private IP with the public one.

Do you have a better approach?

Reply:
I think this will help: https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-alpha/#cmd-phase-kubeconfig . But I'm not sure when it was introduced.